integration/openshift/e2e/runner.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+set -e
+
+GOPATH=${GOPATH:-$(go env GOPATH)}
+script_dir="$(realpath $(dirname $0))"
+source "$script_dir/../../../lib/common.bash"
+tests_dir="$script_dir/../../../"
+source "$script_dir/../lib.sh"
+
+cmd="openshift-tests"
+readonly test_cases_file="${script_dir}/test_cases.txt"
+readonly default_kata_runtimeclass="$kata_runtimeclass"
+
+# Whether to run openshift-tests on dry-run mode or not.
+dry_run="false"
+# Whether to configure the OpenShift cluster or not.
+configure_cluster="false"
+
+function usage() {
+	cat << EOF
+This script runs a sub-set of the OpenShift e2e tests in other to test Kata
+Containers in a running OpenShift cluster.
+
+It assumes that the KUBECONFIG variable is exported and it points to a cluster
+where Kata Containers is already installed.
+
+The cluster also should have:
+ 1) The $default_kata_runtimeclass runtimeClass resource;
+ 2) A running admission controller which annotates any created Pod
+    to use that runtimeClass.
+
+If you choose to have the cluster configured by this script, beware that it
+will not return to its original state on the end.
+
+Usage $0: [-c] [-d] [-r NAME], where:
+  -h, --help   Print this message.
+  -c           Configure the cluster to run the tests.
+  -d           Run the tests in dry-run mode.
+  -r           The runtimeClass name (default is '$default_kata_runtimeclass').
+               This option should not be used with -c.
+EOF
+}
+
+[ -n "$KUBECONFIG" ] || die "The KUBECONFIG variable is unset"
+
+while getopts "hcdr:" opt; do
+	case ${opt} in
+		c)
+			configure_cluster="true"
+			;;
+		d)
+			dry_run="true"
+			;;
+		h)
+			usage
+			exit 0
+			;;
+		r)
+			kata_runtimeclass=${OPTARG}
+			;;
+		*)
+			usage
+			exit 1
+			;;
+	esac
+done
+
+# If the cluster is already configured by the user then only check that the
+# admission controller is active.
+#
+if [ "$configure_cluster" == "true" ]; then
+	[ "$default_kata_runtimeclass" == "$kata_runtimeclass" ] || \
```

Comment:
What is the point of that test? I seem to remember a recent discussion regarding alternate runtime classes, e.g. `kata-qemu-virtiofs-xattr-2.0-softmmu-relaxed-nowarnings`.

Reply:
I made this OpenShift e2e tests support thinking in two use cases:

1. Manual testing. The person should know what she/he is doing, is likely the cluster is already configured but, to be sure, the `runner.sh` is going to do some checking. The user might have configured a `runtimeClass` with, e.g., `my-kata-runtime` and this is going to break the checking. So It was introduced the `-r` option, allowing to overwrite the default `runtimeClass` name used on the check.
1. Automated testing. In this case "the thing" (likely an CI) doesn't have the cluster full configured, and here comes the `-c` option to rescue. It is used the http://github.com/kata-containers/tests/tree/master/kata-webhook code to configure the admission controller, which has hard-coded the `runtimeClass` name to `kata`.

Therefore, if someone passes `-c` and `-r` options together then the configuration step is going to use the default `runtimeClass` name (i.e. `kata`)  and the checking is going to use whatever was passed by the user, and will fail. That's the reason why I added that test.

Makes sense?

Reply:
It does. I would expand the error message then:

```
die "Configuring the cluster uses kata-webhook, which deploys only the default runtimeClass $default_kata_runtime_class, so unable to test with $runtime_class"
```

It's a bit verbose, but at least you know where to look.

Reply:
cool, I will improve the message. Thanks @c3d